aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1104,18 +1112,19 @@ impl AptosVM {
         gas_meter: &mut impl AptosGasMeter,
     ) -> (VMStatus, VMOutput) {
         // Revalidate the transaction.
-        let mut session = self.0.new_session(resolver, SessionId::prologue(txn));
-        if let Err(err) = self.validate_signature_checked_transaction(
+        let txn_data = TransactionMetadata::new(txn);
+        let mut session = self
+            .vm_impl
+            .new_session(resolver, SessionId::prologue_meta(&txn_data));
+        unwrap_or_discard!(self.validate_signed_transaction(
```

Comment:
hmm, is this better than just use .unwrap_or_else(discard_error_vm_status)?;

Reply:
No preference from my side, just reused the existing macro. I guess using `unwrap_or_else()?` is more explicit compared to macro